components/server/src/code-sync/code-sync-service.ts

```diff
@@ -253,24 +252,12 @@ export class CodeSyncService {
                 if (response.status !== 200) {
                     throw new Error(`code sync: blob service: upload failed with ${response.status} ${response.statusText}`);
                 }
-                oldObject = oldRev && toObjectName(resourceKey, oldRev);
+                // delete old revs from storage
+                this.deleteObjects(userId, resourceKey, oldRevs).then().catch(e => {
```

Comment:
this part should be executed only after successful insert transaction, not before, otherwise it can happen that we delete data but not insert

Reply:
change DB insert and doInsert order can resolve

Reply:
I don't think so, since we call from within transaction still, no?

Reply:
Yes, we trust that the transaction commit will succeed so we put the object to GCP, and we delete objects in this step too. (I added a commit now)

We need to resolve problems if things we trust don't do like we expected, that's why we need some GC things, maybe workspace images also have some leak

Reply:
why cannot we just move out of transaction before, so we don't trust, but know for sure?